python/polycap.pyx

```diff
@@ -532,25 +532,39 @@ cdef tuple vector2tuple(polycap_vector3 vec):
 '''Class containing information about the simulated leak events such as position and direction, energy and transmission weights.
 '''
 cdef class Leak:
-    cdef polycap_leak *leak
     cdef object coords
     cdef object direction
     cdef object elecv
     cdef object weight
     cdef int64_t n_refl
 
     def __cinit__(self):
-        self.leak = NULL
         self.coords = None
         self.direction = None
         self.elecv = None
         self.weight = None
         self.n_refl = 0
 
-    def __dealloc__(self):
-        '''Free a :ref:``Leak`` class and all associated data'''
-        if self.leak is not NULL:
-            polycap_leak_free(self.leak)
+    @staticmethod
+    cdef create(polycap_leak *leak):
+        if leak == NULL:
+            return None
+        rv = Leak()
+
+        rv.coords = vector2tuple(leak.coords)
+        rv.direction = vector2tuple(leak.direction)
+        rv.elecv = vector2tuple(leak.elecv)
+
+        cdef np.npy_intp dims[1]
+        dims[0] = leak.n_refl # is this correct??
```

Comment:
helaas niet correct. `n_refl` is hoeveel keer het foton reflecteerde in het optiek
Hier moeten we eigenlijk `photon.n_energies` hebben. Dus ik vermoed dat we naast een polycap_leak *leak ook nog een aantal energies moeten meegeven...
Of ik moet de C struct hervormen zodat die n_energies ook in de Leak struct zelf zit (wat mss meer steek houdt...)

What think you?

Reply:
Yep, best dat die informatie ook in de `polycap_leak` struct zit.

Reply:
Hey Pieter, moet dit nog gebeuren?

Reply:
Die `n_energies` zit nu ook in de leak structure.
Ben momenteel aan het werken om de leak events te kunnen uitlezen in Python, maar ben wat sidetracked omdat ik de nota zag over `vector2tuple` en dat die best een namedtuple zou returnen. Maar slaag er nu niet echt in om de `assertIsInstance` tests niet te laten falen.

Reply:
Commit wat ge hebt en push naar GitHub. Zal er eens naar kijken dan.